beacon-chain/sync/subscriber.go

```diff
@@ -456,6 +479,24 @@ func (s *Service) subscribeStaticWithSyncSubnets(topic string, validator pubsub.
 				if s.chainStarted.IsSet() && s.cfg.InitialSync.Syncing() {
 					continue
 				}
+				valid, err := isDigestValid(digest, genesis, genRoot)
```

Comment:
Tests for here as well

Reply:
This whole method is difficult to measure as it is ticker based, so race conditions are abound. Also it involves tricky logic with libp2p for testing. I want to refactor away this whole file later so testing is easier to carry out here.